src/client/app/redux/selectors/adminSelectors.ts

```diff
@@ -0,0 +1,309 @@
+import { selectConversionsDetails } from '../../redux/api/conversionsApi'
+import { selectAllGroups } from '../../redux/api/groupsApi'
+import { selectAllMeters, selectMeterById } from '../../redux/api/metersApi'
+import { PreferenceRequestItem } from '../../types/items'
+import { ConversionData } from '../../types/redux/conversions'
+import { MeterData, MeterTimeSortType } from '../../types/redux/meters'
+import { UnitData, UnitType } from '../../types/redux/units'
+import { unitsCompatibleWithUnit } from '../../utils/determineCompatibleUnits'
+import { AreaUnitType } from '../../utils/getAreaUnitConversion'
+import { noUnitTranslated, potentialGraphicUnits } from '../../utils/input'
+import translate from '../../utils/translate'
+import { selectAllUnits, selectUnitDataById } from '../api/unitsApi'
+import { selectAdminState } from '../slices/adminSlice'
+import { selectVisibleMetersAndGroups } from './authVisibilitySelectors'
+import { createAppSelector } from './selectors'
+
+export const selectAdminPreferences = createAppSelector(
+	[selectAdminState],
+	(adminState): PreferenceRequestItem => ({
+		displayTitle: adminState.displayTitle,
+		defaultChartToRender: adminState.defaultChartToRender,
+		defaultBarStacking: adminState.defaultBarStacking,
+		defaultLanguage: adminState.defaultLanguage,
+		defaultTimezone: adminState.defaultTimezone,
+		defaultWarningFileSize: adminState.defaultWarningFileSize,
+		defaultFileSizeLimit: adminState.defaultFileSizeLimit,
+		defaultAreaNormalization: adminState.defaultAreaNormalization,
+		defaultAreaUnit: adminState.defaultAreaUnit,
+		defaultMeterReadingFrequency: adminState.defaultMeterReadingFrequency,
+		defaultMeterMinimumValue: adminState.defaultMeterMinimumValue,
+		defaultMeterMaximumValue: adminState.defaultMeterMaximumValue,
+		defaultMeterMinimumDate: adminState.defaultMeterMinimumDate,
+		defaultMeterMaximumDate: adminState.defaultMeterMaximumDate,
+		defaultMeterReadingGap: adminState.defaultMeterReadingGap,
+		defaultMeterMaximumErrors: adminState.defaultMeterMaximumErrors,
+		defaultMeterDisableChecks: adminState.defaultMeterDisableChecks,
+		defaultHelpUrl: adminState.defaultHelpUrl
+	})
+)
+
+
+export const selectPossibleGraphicUnits = createAppSelector(
+	selectUnitDataById,
+	unitDataById => potentialGraphicUnits(unitDataById)
+)
+
+/**
+ * Calculates the set of all possible meter units for a meter.
+ * This is any unit that is of type unit or suffix.
+ * @returns The set of all possible graphic units for a meter
+ */
+export const selectPossibleMeterUnits = createAppSelector(
+	selectAllUnits,
+	unitData => {
+		let possibleMeterUnits = new Set<UnitData>();
+		// The meter unit can be any unit of type meter.
+		unitData.forEach(unit => {
+			if (unit.typeOfUnit == UnitType.meter) {
+				possibleMeterUnits.add(unit);
+			}
+		});
+		// Put in alphabetical order.
+		possibleMeterUnits = new Set(_.sortBy(Array.from(possibleMeterUnits), unit => unit.identifier.toLowerCase(), 'asc'));
```

Comment:
Thius is not a change but I noticed in src/client/app/redux/api/unitsApi.ts that the sorting includes an accent term and localCompare. There are other places where the way we sort differs. Do you think they should be consistent? If so, we should see if a function(s) can be used to do this in an easier way.

Reply:
Will get consistent in future.